src/lib.rs

```diff
@@ -577,46 +583,65 @@ enum SignAlgo {
 	EdDsa(&'static EdDSAParameters),
 }
 
-enum DynKeyPair {
-	EcKp(EcdsaKeyPair),
-	EdKp(Ed25519KeyPair),
+/// A key pair used to sign certificates and CSRs
+pub enum KeyPair {
+	/// A Ecdsa key pair
+	EcKp(EcdsaKeyPair, Vec<u8>),
+	/// A Ed25519 key pair
+	EdKp(Ed25519KeyPair, Vec<u8>),
+}
+
+impl From<&[u8]> for KeyPair {
+	fn from(pkcs8 :&[u8]) -> KeyPair {
+		use std::iter::FromIterator;
+
+		if let Ok(edkp) = Ed25519KeyPair::from_pkcs8(Input::from(pkcs8)) {
```

Comment:
Could you add a `let input = Input::from(pkcs8);` to the front and then use it in each function? This makes the lines shorter. Possibly you'll have to clone it, most likely you won't.

Reply:
Did that. I also move the pkc8 vector cloning in front: https://github.com/est31/rcgen/commit/33ea827ce1dcbee50d88dea2adfd8b4c548dbc2c#diff-b4aea3e418ccdb71239b96952d9cddb6R591